### PR TITLE
Update Build Process

### PR DESCRIPTION
**Pull request by Austio**
Pivotal Link: https://www.pivotaltracker.com/story/show/139238377

This PR Updates multiple areas of the app to allow us to run a build script to generate package that is used by doximity-client-vue.  Some items of note

1. We are not minifying right now, this will allow us to live edit in our node_modules after a build if we want to in order to debug something
2. Build is now for umd, which will target both common and umd
3. "transform-es2015-modules-commonjs" - we needed this in babel because of an issue related to something assigning values to module exports `module.exports = 'foo'`

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,5 +1,11 @@
 {
-  "presets": ["es2015", "stage-2"],
-  "plugins": ["transform-runtime"],
+  "presets": [
+    ["es2015", { "modules": false }],
+    "stage-2"
```
Comment (aurelienbottazini) [inline .babelrc]: are you ok with removing stage2 and adding es2016, es2017 and object spread? i think it is clearer (more declarative). If yes i can do it with another PR :)

File: build/webpack.dist.conf.js [hunk 2]
```diff
@@ -9,55 +9,39 @@ var env = process.env.NODE_ENV === 'testing'
...
     // http://vuejs.github.io/vue-loader/en/workflow/production.html
     new webpack.DefinePlugin({
       'process.env': env
     }),
-    new webpack.optimize.UglifyJsPlugin({
-      compress: {
-        warnings: false
-      },
-      sourceMap: config.build.productionSourceMap
+    // new webpack.optimize.UglifyJsPlugin({
```
Comment (aurelienbottazini) [inline build/webpack.dist.conf.js]: why did you remove it? is it done elsewhere?
Or optimizations are WIP?

File: src/index.js [hunk 3]
```diff
@@ -1,14 +1,18 @@
-import Components from './components';
+import components from './components';
 
-function genomeName(key) {
-  return `vg-${key.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase()}`;
-}
+const kebobCase = str => str.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase();
```
Comment (aurelienbottazini) [inline src/index.js]: typo? should it be "kebabCase" ? I am not sure what is kebob case 😄

Comment (Austio) [inline src/index.js]: haha

File: build/webpack.dist.conf.js (hunk 2 above)
Comment (Austio) [inline build/webpack.dist.conf.js]: Removing the Uglify so that we can see/edit the code in place in the client in either node_modules or inline.  Adding will just be commenting when we want to deploy real version

File: .babelrc (hunk 1 above)
Comment (Austio) [inline .babelrc]: Yeah, i don't have strong opinion on it, will switch in next push

File: src/index.js (hunk 3 above)
Comment (Austio) [inline src/index.js]: Fixed!

File: build/webpack.dist.conf.js (hunk 2 above)
Comment (aurelienbottazini) [inline build/webpack.dist.conf.js]: ah yes you did say that to me i forgot...

File: .babelrc (hunk 1 above)
Comment (Austio) [inline .babelrc]: note: ci failure on this so will try in future pr
